types/store.go

```diff
@@ -287,7 +287,7 @@ func (key *KVStoreKey) String() string {
 // range query for all []byte with a certain prefix
 // Deals with last byte of prefix being FF without overflowing
 func PrefixEndBytes(prefix []byte) []byte {
-	if prefix == nil {
+	if prefix == nil || len(prefix) == 0 {
```

Comment:
I think `if len(prefix) == 0` will be enough, because  `len(nil)` will be zero as well.